### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by rudolfs**
I ran `npx npm-check -u`.
 
  - upgrade patch level deps
  - upgrade minor level deps
  - upgrade major level deps
  - upgrade to node v18

Didn't upgrade the gnosis deps, because I didn't want to invest time on deprecated orgs stuff:
```
@gnosis.pm/safe-core-sdk        0.3.1  ❯  3.1.1  https://github.com/safe-global/safe-core-sdk#readme
@gnosis.pm/safe-core-sdk-types  0.1.1  ❯  1.6.1  https://github.com/safe-global/safe-core-sdk#readme
@gnosis.pm/safe-service-client  0.1.1  ❯  1.3.1  https://github.com/safe-global/safe-core-sdk#readme
```

⚠️ Important notes: ⚠️
 - the [Ceramic upgrade](https://github.com/radicle-dev/radicle-interface/pull/401/commits/d6fc5c6bba31f9da2a9048eef8bc015a10d2fa28) required bumping our build target to es2020
 - after the [vite upgrade](https://github.com/radicle-dev/radicle-interface/pull/401/commits/b0e1d61a67041d9bcf28cbcad536eb053d13103d#diff-6a3b01ba97829c9566ef2d8dc466ffcffb4bdac08706d3d6319e42e0aa6890ddR48) i had to bind the local dev server to `127.0.0.1` specifically, otherwise CI would fail, I assume it's because it would otherwise bind to the ipv6 address
 - this needs extensive testing

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export default defineConfig({
+  video: false,
+  defaultCommandTimeout: 10000,
+  e2e: {
+    specPattern: "cypress/e2e/**/*spec.ts",
+    baseUrl: "http://localhost:3000",
+  },
+  component: {
+    devServer: {
+      framework: "svelte",
```
Comment (sebastinez) [inline cypress.config.ts:12]: Nice! That you found this update! 🎉  💪

Comment (sebastinez) [reply cypress.config.ts:12]: Just for reference it seems that component testing now also works with node v17

File: src/SeedAddress.spec.ts [hunk 2]
```diff
@@ -1,10 +1,8 @@
 import SeedAddress from "./SeedAddress.svelte";
-import { render } from "@testing-library/svelte";
-import "@public/index.css";
 
 describe("SeedAddress", () => {
   it("shows the seed emoji and seed host", () => {
-    render(SeedAddress, {
+    cy.mount(SeedAddress, {
       props: {
         seed: {
```
Comment (sebastinez) [inline src/SeedAddress.spec.ts:7]: I'm not sure why the error hasn't shown up before, but it seems we should cast the `seed` property to the `Seed` class as:
```diff
diff --git a/src/SeedAddress.spec.ts b/src/SeedAddress.spec.ts
index 4e4b4ed..b651967 100644
--- a/src/SeedAddress.spec.ts
+++ b/src/SeedAddress.spec.ts
@@ -1,3 +1,4 @@
+import type { Seed } from "@app/base/seeds/Seed";
 import SeedAddress from "./SeedAddress.svelte";
 
 describe("SeedAddress", () => {
@@ -8,7 +9,7 @@ describe("SeedAddress", () => {
           id: "hydkkkf5ksbe5fuszdhpqhytu3q36gwagj874wxwpo5a8ti8coygh1",
           emoji: "🐱",
           host: "seed.cloudhead.io",
-        },
+        } as Seed,
         port: 8776,
       },
     });
@@ -25,7 +26,7 @@ describe("SeedAddress", () => {
           id: "hydkkkf5ksbe5fuszdhpqhytu3q36gwagj874wxwpo5a8ti8coygh1",
           emoji: "🐱",
           host: "seed.cloudhead.io",
-        },
+        } as Seed,
         port: 8776,
         full: true,
       },
```

**Review by sebastinez — COMMENTED** [review]
So far looking nice!
Kudos again for updating the the component testing library to work directly with the Cypress v10 config, didn't realize this came out, so much nicer!
Also the changes to the ordering of imports is nice 👍 

Left one comment that threw a compiler error on my system, maybe we can add that

Will have to investigate also the changes in the Gnosis SDKs.

**Comment by sebastinez** [comment]
Just tried running the app and it seems there is an issue with `es2020` not working on vite if don't specified in the `esbuild` options
Reference: https://github.com/vitejs/vite/issues/9062

We should add this to `vite.config.ts`
```diff
diff --git a/vite.config.ts b/vite.config.ts
index 2d45fd7..2477600 100644
--- a/vite.config.ts
+++ b/vite.config.ts
@@ -9,6 +9,9 @@ import type { Request, Response } from "express-serve-static-core";
 
 const config: UserConfig = {
   optimizeDeps: {
+    esbuildOptions: {
+      target: "es2020",
+    },
     exclude: [
       "svelte-routing",
       "@pedrouid/environment",
```

File: src/SeedAddress.spec.ts (hunk 2 above)
Comment (rudolfs) [reply src/SeedAddress.spec.ts:7]: It's weird that it wasn't caught by our CI. Will look into it.

**Comment by rudolfs** [comment]
TODO: fix `browser-external:buffer:9 Module "buffer" has been externalized for browser compatibility. Cannot access "buffer.Buffer" in client code.`

**Comment by rudolfs** [comment]
> Just tried running the app and it seems there is an issue with `es2020` not working on vite if don't specified in the `esbuild` options Reference: [vitejs/vite#9062](https://github.com/vitejs/vite/issues/9062)
> 
> We should add this to `vite.config.ts`
> 
> ```diff
> diff --git a/vite.config.ts b/vite.config.ts
> index 2d45fd7..2477600 100644
> --- a/vite.config.ts
> +++ b/vite.config.ts
> @@ -9,6 +9,9 @@ import type { Request, Response } from "express-serve-static-core";
>  
>  const config: UserConfig = {
>    optimizeDeps: {
> +    esbuildOptions: {
> +      target: "es2020",
> +    },
>      exclude: [
>        "svelte-routing",
>        "@pedrouid/environment",
> ```

Hmm, was there an error somewhere? It seemed to work for me. I did have to `rm -rf node_modules && npm install` thought.

File: cypress.config.ts (hunk 1 above)
Comment (rudolfs) [reply cypress.config.ts:12]: Are you referring to https://github.com/radicle-dev/radicle-interface/commit/d49130163d04eed45fbf48d5f572e38d11c06654?

**Comment by sebastinez** [comment]
> > Just tried running the app and it seems there is an issue with `es2020` not working on vite if don't specified in the `esbuild` options Reference: [vitejs/vite#9062](https://github.com/vitejs/vite/issues/9062)
> > We should add this to `vite.config.ts`
> > ```diff
> > diff --git a/vite.config.ts b/vite.config.ts
> > index 2d45fd7..2477600 100644
> > --- a/vite.config.ts
> > +++ b/vite.config.ts
> > @@ -9,6 +9,9 @@ import type { Request, Response } from "express-serve-static-core";
> >  
> >  const config: UserConfig = {
> >    optimizeDeps: {
> > +    esbuildOptions: {
> > +      target: "es2020",
> > +    },
> >      exclude: [
> >        "svelte-routing",
> >        "@pedrouid/environment",
> > ```
> 
> Hmm, was there an error somewhere? It seemed to work for me. I did have to `rm -rf node_modules && npm install` thought.

Weird after doing other stuff an coming back to it, removing `node_modules` again which I'm sure I did the first time, the error went away..

The error I had was a lot of bundled code filling up my console with `Big integer literals are not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari13" + 2 overrides)``
Will let you know if it shows up again.

File: cypress.config.ts (hunk 1 above)
Comment (sebastinez) [reply cypress.config.ts:12]: Didn't think about that commit, just for local development, but I think we could also remove that version restriction

**Comment by sebastinez** [comment]
> > Hmm, was there an error somewhere? It seemed to work for me. I did have to `rm -rf node_modules && npm install` thought.

Ok it seems to break when updating `@self.id/core` to v0.4.1, which updates `key-did-resolver` from v1.4.4 to v2.1.2.

**Comment by sebastinez** [comment]
Ok just tried out
```
 Non-Semver Versions less than 1.0.0, caution.
 ◯ @datamodels/identity-profile-basic  0.1.2  ❯  0.2.0  https://github.com/ceramicstudio/datamodels#readme
 ◯ @self.id/core                       0.1.0  ❯  0.4.1  https://github.com/ceramicstudio/self.id#readme
```
and except the issue at https://github.com/radicle-dev/radicle-interface/pull/401#issuecomment-1256331079 it seems like IDX still keeps working fine

**Comment by rudolfs** [comment]
> > > Just tried running the app and it seems there is an issue with `es2020` not working on vite if don't specified in the `esbuild` options Reference: [vitejs/vite#9062](https://github.com/vitejs/vite/issues/9062)
> > > We should add this to `vite.config.ts`
> > > ```diff
> > > diff --git a/vite.config.ts b/vite.config.ts
> > > index 2d45fd7..2477600 100644
> > > --- a/vite.config.ts
> > > +++ b/vite.config.ts
> > > @@ -9,6 +9,9 @@ import type { Request, Response } from "express-serve-static-core";
> > >  
> > >  const config: UserConfig = {
> > >    optimizeDeps: {
> > > +    esbuildOptions: {
> > > +      target: "es2020",
> > > +    },
> > >      exclude: [
> > >        "svelte-routing",
> > >        "@pedrouid/environment",
> > > ```
> > 
> > 
> > Hmm, was there an error somewhere? It seemed to work for me. I did have to `rm -rf node_modules && npm install` thought.
> 
> Weird after doing other stuff an coming back to it, removing `node_modules` again which I'm sure I did the first time, the error went away..
> 
> The error I had was a lot of bundled code filling up my console with `Big integer literals are not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari13" + 2 overrides)` Will let you know if it shows up again.

I ran `npm cache clean --force` and installed the latest node and also got the same error. Applied your patch.

File: cypress.config.ts (hunk 1 above)
Comment (rudolfs) [reply cypress.config.ts:12]: Bumped everything to node 18: 8fc669706162251111a984c3624f9902adf7ccef.

**Comment by rudolfs** [comment]
Weirdly, now it fails on CI 🤔 

https://github.com/radicle-dev/radicle-interface/actions/runs/3114939712/jobs/5051328234

File: src/SeedAddress.spec.ts (hunk 2 above)
Comment (rudolfs) [reply src/SeedAddress.spec.ts]: 0bbdccdbe019377314ee4e3278f602d2e9399b32

**Comment by rudolfs** [comment]
Superseded by https://github.com/radicle-dev/radicle-interface/pull/427.

**Pull request closed without merging**